### ISSUE TITLE
SCOPE environment variable set causing db:migrate to do nothing 

### ISSUE THREAD

**Issue by cklatt** [rails/rails#41100]
Our rails application set an environment variable named SCOPE.  This was defined in our application.yml file and loaded by Figaro.  Some time later we could not figure out why none of our database migrations would work.  We spent over 3 hours debugging and trying to figure out what happened.  

With the SCOPE environment variable defined running rake db:migrate would do the following:

myself@pc-admin % rake db:migrate
myself@pc-admin %

This appeared as if db:migrate failed silently.  Running rake db:migrate:status showed us which migrations needed to be run.  rake db:migrate --verbose --trace and other flags did not reveal any additional clues as to what was happening.

Finally, we read the code and found that migrate uses the value of ENV["SCOPE"]

```
 def migrate
    check_target_version

    scope = ENV["SCOPE"]
    ....
```

I propose a small change that will output the value of SCOPE if defined as an environment variable.  That probably would have been a sufficient clue to identify the conflict.

Something like:

`    puts "\nmigration scope set from ENV["SCOPE"] to value of #{ENV["SCOPE"]}" if ENV.has_key?("SCOPE")`

I think this should be displayed regardless of verbose setting, but could see it that way as well.

**Comment by intrip** [comment]
@gmcgibbon what about extending https://github.com/rails/rails/blob/main/activerecord/lib/active_record/tasks/database_tasks.rb so that if `SCOPE` is set and no migration ran it will print a message? I was thinking of something like `No migrations found for #{ENV["SCOPE"]} scope.`

If we agree on an implementation I'd be happy to make a PR

**Comment by cklatt** [comment]
@intrip I think that will work.  The only add I'd like to see is to incorporate the source ENV['SCOPE'] in the message so the user fully understands where the value of SCOPE came from.  Another option is to output a more detailed message such as "SCOPE = value #{ENV['SCOPE']} from ENV['SCOPE']" when verbose flag is used.   Either way, I think your fix is a good idea.

**Comment by intrip** [comment]
@cklatt thanks! 
I'll wait for the input from @gmcgibbon as well before starting so that I don't work on something which will eventually be rejected 😄

**Comment by gmcgibbon** [comment]
> This appeared as if db:migrate failed silently.

I agree. I think it would be a good change to give feedback to the user regardless of the environment that says "No migrations ran." or something along those lines. One thing to note here is that if we log here we need to do the same in rollback so the experience is consistent across commands. I believe this also justifies moving rollback to `DatabaseTasks`.

> I propose a small change that will output the value of SCOPE if defined as an environment variable.

I think appending it to migrate's no-op message would be acceptable (eg. "No migrations ran. (using SCOPE scope)"). 


@intrip let me know if you are still interested in working on this.

**Comment by intrip** [comment]
@gmcgibbon I am still interested. Will work on a PR.

**Comment by intrip** [comment]
@gmcgibbon I've created #41234.

I'd like some feedback:
- The PR introduces a warning only for the `migrate` command, because `ENV["SCOPE"]` [only affects migrate command](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/tasks/database_tasks.rb#L271-L284).
- I'm completely open in extending `SCOPE` in order to work with the other commands for consistency sake (e.g `rollback, forward`...) but I'm not sure if we should do it at all because it could break compatibility and might lead into undesired behaviour; moreover this could be handled in a separate PR/Issue.
